frontend/src/main/components/Jobs/UpdateCoursesQuarterJobForm.js

```diff
@@ -0,0 +1,66 @@
+import { useState } from "react";
+import { Form, Button, Container, Row, Col } from "react-bootstrap";
+
+import { quarterRange } from "main/utils/quarterUtilities";
+
+import { useSystemInfo } from "main/utils/systemInfo";
+import SingleQuarterDropdown from "../Quarters/SingleQuarterDropdown";
+import { useBackend  } from "main/utils/useBackend";
+
+const UpdateCoursesQuarterJobForm = ({ callback }) => {
+
+  const { data: systemInfo } = useSystemInfo();
+
+  // Stryker disable OptionalChaining
+  const startQtr = systemInfo?.startQtrYYYYQ;
+  const endQtr = systemInfo?.endQtrYYYYQ;
+  // Stryker enable OptionalChaining
+
+  const quarters = quarterRange(startQtr, endQtr);
+
+  // Stryker disable all : not sure how to test/mock local storage
+  const localQuarter = localStorage.getItem("BasicSearch.Quarter");
+
+  const { data: _subjects, error: _error, status: _status } =
+  useBackend(
+    // Stryker disable next-line all : don't test internal caching of React Query
+    ["/api/UCSBSubjects/all"], 
+    { method: "GET", url: "/api/UCSBSubjects/all" }, 
+    []
+  );
+
+  const [quarter, setQuarter] = useState(localQuarter || quarters[0].yyyyq);
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    console.log("UpdateCoursesJobForm: quarter", quarter);
+    callback({ quarter });
+  };
+
+  // Stryker disable all : Stryker is testing by changing the padding to 0. But this is simply a visual optimization as it makes it look better
```

Comment:
It's interesting to see this text show up in both the 5pm and 7pm forks of the project... did you collaborate on this?

In any case: here's the refactor to limit the scope of Stryker exlcusions.  We try to keep these to a minimum so they don't end up hiding other problems:

https://github.com/ucsb-cs156-f22/f22-5pm-courses/pull/57/files

Reply:
That comment came from UpdateCoursesJobForm.js because I just copied that file to create UpdateCoursesQuarterJobForm.js